lib/Doctrine/Annotations/Type/Constant/ConstantFloatType.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Type\Constant;
+
+use Doctrine\Annotations\Type\ConstantScalarType;
+use Doctrine\Annotations\Type\FloatType;
+use function abs;
+use function is_float;
+use function sprintf;
+
+/**
+ * @internal
+ */
+final class ConstantFloatType extends FloatType implements ConstantScalarType
+{
+    public const EPSILON = 1e-15;
+
+    /** @var float */
+    private $value;
+
+    public function __construct(float $value)
+    {
+        $this->value = $value;
+    }
+
+    public function getValue() : float
+    {
+        return $this->value;
+    }
+
+    public function describe() : string
+    {
+        return sprintf('%F', $this->value);
```

Comment:
I would sugget to use just `(string) $this-value`
```php
>>> sprintf('%F', 3.3)
=> "3.300000"
>>> (string) 3.3
=> "3.3"
```

Reply:
No: https://3v4l.org/s1FrW

Reply:
better than
```php
>>> sprintf('%F', 3.33333333333333333333333333333333333333334)
=> "3.333333"`
```

Reply:
I'd agree if used for validation, but here we're talking about describe() where simpleness is preferable. So keeping as is.